experiment/simulation/js/mild.js

```diff
@@ -0,0 +1,237 @@
+            [plateStartX + tubeWidth2, bottomPlateStartY - tubeHeight1 - tubeHeight2],
+            [plateStartX + plateWidth1 - tubeWidth2, bottomPlateStartY - tubeHeight1 - tubeHeight2],
+            [plateStartX + plateWidth1 - tubeWidth, bottomPlateStartY - tubeHeight1],
+            [plateStartX + plateWidth1 - tubeWidth, bottomPlateStartY],
+
+        ];
+        step = 0;
+
+        document.getElementById("stress").innerHTML = "0.0 Mpa";
+        document.getElementById("strain").innerHTML = "0.0";
+        document.getElementById("young").innerHTML = "0.0 Mpa";
+        document.getElementById("load").innerHTML = "0.0 N";
+        document.getElementById("elongation").innerHTML = "0.0 mm";
+
+    }
+
+    function restart() {
+        window.clearTimeout(tmHandle);
+        setAll();
+        graph();
+        play();
+    }
+
+    function play() {
+        tmHandle = window.setTimeout(draw, 1000 / fps);
+        pauseButton.removeAttribute("disabled");
+        restartButton.removeAttribute("disabled");
+        playButton.setAttribute("disabled", "true");
+    }
+
+    function pause() {
+        window.clearTimeout(tmHandle);
+        pauseButton.setAttribute("disabled", "true");
+        playButton.removeAttribute("disabled");
+    }
+
+    function drawObject(ctx, obj, color) {
+        ctx.save();
+        ctx.fillStyle = color;
+        ctx.strokeStyle = color;
+        ctx.beginPath();
+        ctx.moveTo(obj[0][0], obj[0][1]);
+
+        for (let i = 0; i < obj.length; ++i) {
+            const next = (i + 1) % obj.length;
+            ctx.lineTo(obj[next][0], obj[next][1]);
+        }
+
+        ctx.closePath();
+        ctx.fill();
+        ctx.stroke();
+        ctx.restore();
+    }
+
+    function move(tube, flag) {
+        for (let i = 0; i < tube.length; ++i) {
+            tube[i][1] += flag;
+        }
+    }
+
+    let topPlate = [];
+    let bottomPlate = [];
+    let upperTube = [];
+    let lowerTube = [];
+
+
+    const canvas = document.getElementById("main");
+    canvas.width = 900;
+    canvas.height = 1000;
+    canvas.style = "border:3px solid;";
+    const ctx = canvas.getContext("2d");
+
+    const fill = "#A9A9A9";
+    const lineWidth = 1.5;
+    const originalFPS = 20;
+    let fps = 20;
+
+
+    const plateStartX = 250;
+    const plateWidth1 = 400;
+    const plateWidth2 = 300;
+    const plateStartY = 150;
+    const bottomPlateStartY = 750;
+    const plateHeight1 = 100;
+    const plateHeight2 = 60;
+    const tubeWidth = 240;
+    const tubeWidth2 = 215;
+    const tubeHeight1 = 220;
+    const tubeHeight2 = 35;
+    let step = 0;
+    let chart = [];
+
+
+
+    setAll();
+    drawStatic();
+    graph();
+    let tmHandle;
+
+    function drawStatic() {
+        ctx.clearRect(0, 0, canvas.width, canvas.height);
+        ctx.fillStyle = fill;
+        ctx.lineWidth = lineWidth;
+        ctx.lineCap = "round";
+        ctx.lineJoin = "round";
+        ctx.fillStyle = "black";
```

Comment:
Please move this to data.colors.
We are assigning it twice, lines 169 and 173. Please remove one